packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -737,6 +739,31 @@ abstract class ResidentRunner {
     throw Exception('Canvaskit not supported by this runner.');
   }
 
+  /// Dump the sksl shaders to a file in the project directory.
+  Future<void> writeSkSl() async {
+    if (!supportsWriteSkSl) {
+      throw Exception('writeSkSl is not supported by this runner.');
+    }
+    final Map<String, Uint8List> data = await flutterDevices.first.views.first.getSkSls();
+    if (data.isEmpty) {
+      globals.logger.printStatus(
+        'No data was receieved. To ensure SkSl data can be generated: \n'
+        '  1. Pass "--cache-sksl" as an argument to flutter run.\n'
+        '  2. Interact with the application to force shaders to be compiled.\n'
+      );
+      return;
+    }
+    final Directory outputDirectory = globals.fs.currentDirectory
+      .childDirectory('shaders')
```

Comment:
Maybe use directory `shaders/sksl` or just `sksl` to be more specific. There's a possibility that we can have other kind of shader artifacts (e.g., binary shaders).

Reply:
Also, maybe put the auto-captured shaders under a folder with flutter version in it (e.g., `shaders/8857c4cec89164fdb6293bef69f9d62c62733efa/sksl`) because `SkSL` is not guaranteed to be backward compatible. A Flutter app should use the SkSLs that are captured from the same framework/engine version.

Reply:
Good to know! Yeah, we can version these based on the specific engine hash that flutter is built with and refuse to include older shaders